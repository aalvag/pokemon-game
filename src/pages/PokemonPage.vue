<template>
  <h1>Who is that Pok√©mon?</h1>
  <PokemonPicture :pokemonId="66" :showPokemon="false" />
  <PokemonOptions :pokemons="pokemonArr" />
</template>

<script>
import PokemonOptions from "@/components/PokemonOptions.vue";
import PokemonPicture from "@/components/PokemonPicture.vue";
import { getPokemonOptions } from "@/helpers/getPokemonOptions";

export default {
  name: "PokemonPage",
  components: { PokemonPicture, PokemonOptions },
  data() {
    return {
      pokemonArr: [],
    };
  },
  methods: {
    async mixesPokemons() {
      const pokemons = await getPokemonOptions();
      this.pokemonArr = pokemons;
    },
  },
  mounted() {
    this.mixesPokemons();
  },
};
</script>

<style lang="scss" scoped></style>
