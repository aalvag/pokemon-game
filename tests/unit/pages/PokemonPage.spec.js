import PokemonPage from "@/pages/PokemonPage.vue";
import { shallowMount } from "@vue/test-utils";
import { pokemonsMock } from "../mocks/pokemons.mock";

describe("PokemonPage", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("must match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("calls getPokemonOptions when the component is created", () => {
    const mixesPokemonsSpy = jest.spyOn(PokemonPage.methods, "mixesPokemons");
    shallowMount(PokemonPage);
    expect(mixesPokemonsSpy).toHaveBeenCalled();
  });

  test("must match the snapshot when load pokemons", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonsMock,
          pokemon: pokemonsMock[0],
          showPokemon: false,
          message: "",
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("must show the component PokemonPicture and PokemonOptions", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonsMock,
          pokemon: pokemonsMock[0],
          showPokemon: false,
          message: "",
        };
      },
    });

    const pokemonPicture = wrapper.find("pokemon-picture-stub");
    const pokemonOptions = wrapper.find("pokemon-options-stub");

    expect(pokemonPicture.exists()).toBe(true);
    expect(pokemonOptions.exists()).toBe(true);

    expect(pokemonPicture.attributes("pokemonid")).toBe("1");
    expect(pokemonOptions.attributes("pokemons")).toBeTruthy();
  });

  test("test the method checkPokemon", async () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonsMock,
          pokemon: pokemonsMock[0],
          showPokemon: false,
          message: "",
        };
      },
    });

    await wrapper.vm.checkPokemon(1);

    expect(wrapper.find("h2").exists()).toBe(true);
    expect(wrapper.vm.message).toBe(`Correct, is ${pokemonsMock[0].name}`);
    expect(wrapper.vm.showPokemon).toBe(true);

    await wrapper.vm.checkPokemon(20);

    expect(wrapper.vm.message).toBe("Oh, it's not correct");
  });
});
