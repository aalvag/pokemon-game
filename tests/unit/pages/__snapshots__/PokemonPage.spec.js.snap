// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage must match the snapshot 1`] = `
<div>
  <h1>Who is that Pokémon?</h1>
  <h1>Loading...</h1>
  <!--v-if-->
</div>
`;

exports[`PokemonPage must match the snapshot when load pokemons 1`] = `
<div>
  <h1>Who is that Pokémon?</h1>
  <div>
    <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
    <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  </div>
  <!--v-if-->
</div>
`;
